inference-engine/src/legacy_api/src/transformations/convert_opset1_to_legacy/convert_nms_5_to_legacy.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/graph_util.hpp>
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/opsets/opset5.hpp>
+#include <legacy/ngraph_ops/nms_ie.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <transformations/utils/utils.hpp>
+
+#include "legacy/transformations/convert_opset1_to_legacy/convert_nms_5_to_legacy.hpp"
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ConvertNMS5ToLegacyMatcher, "ConvertNMS5ToLegacyMatcher", 0);
+
+ngraph::pass::ConvertNMS5ToLegacyMatcher::ConvertNMS5ToLegacyMatcher() {
+    auto nms = ngraph::pattern::wrap_type<ngraph::opset5::NonMaxSuppression>();
+
+    ngraph::matcher_pass_callback callback = [](pattern::Matcher &m) {
+        auto nms_5 = std::dynamic_pointer_cast<ngraph::opset5::NonMaxSuppression>(m.get_match_root());
+        if (!nms_5) {
+            return false;
+        }
+
+        const auto new_args = nms_5->input_values();
+        const std::size_t num_of_inputs = new_args.size();
+
+        const auto& arg2 = num_of_inputs > 2 ? new_args.at(2) : ngraph::opset5::Constant::create(element::i32, Shape{}, {0});
+        const auto& arg3 = num_of_inputs > 3 ? new_args.at(3) : ngraph::opset5::Constant::create(element::f32, Shape{}, {.0f});
+        const auto& arg4 = num_of_inputs > 4 ? new_args.at(4) : ngraph::opset5::Constant::create(element::f32, Shape{}, {.0f});
+
+        // vector of new nGraph operations
+        NodeVector new_ops;
+
+        auto one_dim_shape = Shape{1};
+
+        Output<Node> new_max_per_class;
+        Output<Node> new_iou_threshold;
+        Output<Node> new_score_threshold;
+        Output<Node> new_soft_nms_sigma;
+
+        Output<Node> new_shape_for_max_per_class = opset1::Constant::create(ngraph::element::i64, Shape{1}, {1});
+        Output<Node> new_shape_for_iou_threshold = opset1::Constant::create(ngraph::element::i64, Shape{1}, {1});
+        Output<Node> new_shape_for_score_threshold = opset1::Constant::create(ngraph::element::i64, Shape{1}, {1});
+        Output<Node> new_shape_for_soft_nms_sigma = opset1::Constant::create(ngraph::element::i64, Shape{1}, {1});
+
+        new_max_per_class = std::make_shared<opset1::Reshape>(arg2, new_shape_for_max_per_class, true);
+        new_ops.push_back(new_max_per_class.get_node_shared_ptr());
```

Comment:
If we write `new_max_per_class` instead of `new_max_per_class.get_node_shared_ptr()`, we get the compilation error, because ` NodeVector` is defined as `using NodeVector = std::vector<std::shared_ptr<Node>>;` at line 29 of `./ngraph/core/include/ngraph/output_vector.hpp`, not as `std::vector<Output<Node>>`. 

And moreover, it is not true that `new_max_per_class == new_max_per_class.get_node_shared_ptr()`, because `new_max_per_class` has type `Output<Node>`, ` new_max_per_class.get_node_shared_ptr()` has type `std::shared_ptr<Node>`, and equality comparison for these types is not defined.

Reply:
Hmm, I don't understand why `new_max_per_class` has a type `Output<Node>` because `make_shared` should create a pointer to `opset1::Reshape`

Reply:
Because there are declaration `Output<Node> new_max_per_class` and the constructor of `Output<Node>` with the argument of the type `std::shared_ptr<Node>`.